File: tools/improve/tests/original/maudec/py/int-funs.py
import enum

class EuclidResult(enum.Enum):
	Euc = 0  # int, int, int
	Err = 1  # 


def eeuclid(M: int, U1: int, V1: int, N: int, U2: int, V2: int) -> EuclidResult:
	match N:
		case 0:
			return (EuclidResult.Euc, M, U1, V1)

		case N:
			aux1 : int = M // N

			return eeuclid(N, U2, V2, M - N * aux1, U1 - U2 * aux1, V1 - V2 * aux1)



def eeuclid_1(M: int, N: int) -> EuclidResult:
	aux1 : int = 1

	return eeuclid(M, aux1, 0, N, 0, aux1)

File: tools/improve/tests/original/maudec/rust/int-funs.rs


#[derive(Clone, PartialEq, PartialOrd, Debug)]
enum EuclidResult {
	Euc(i32, i32, i32),
	Err,
}

fn eeuclid(m: i32, u1: i32, v1: i32, n: i32, u2: i32, v2: i32) -> EuclidResult {
	match n {
		0 => EuclidResult::Euc(m, u1, v1),
		n => {
			let aux1 : i32 = m / n;

			eeuclid(n, u2, v2, m - n * aux1, u1 - u2 * aux1, v1 - v2 * aux1)
		},
	}
}

fn eeuclid_1(m: i32, n: i32) -> EuclidResult {
	let aux1 : i32 = 1;
	eeuclid(m, aux1, 0, n, 0, aux1)
}

